### DISCUSSION TITLE
Does Postal use postfix, sendmail or another MTA to deliver the emails?

### DISCUSSION THREAD

**Discussion by collimarco** [postalserver/postal#2711] [Help with using Postal]
Does Postal use postfix, sendmail or another MTA to deliver the emails?

Or does Postal deliver the emails to the recipient directly from Ruby code?

In this case, is the email delivery built-in into Postal code or are you using a specific gem for the SMTP delivery?

**Comment by willpower232** [comment]
Definitely not using postfix, sendmail, or another mta, have a look at https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/lib/postal/smtp_sender.rb

**Reply by collimarco** [reply]
Thanks! It's cool to see a pure Ruby implementation. So basically Postal composes the email content, then simply uses Ruby Net::SMTP for the delivery (and catches the various exceptions).
